Admin/MenuNodeAdmin.php

```diff
@@ -28,14 +42,87 @@ protected function configureFormFields(FormMapper $formMapper)
     {
         $formMapper
             ->with('form.group_general')
-                ->add(
-                    'parent',
-                    'doctrine_phpcr_odm_tree',
-                    array('root_node' => $this->menuRoot, 'choice_list' => array(), 'select_root_node' => true)
-                )
+                ->add('parent', 'doctrine_phpcr_odm_tree', array(
+                    'root_node' => $this->menuRoot,
+                    'choice_list' => array(),
+                    'select_root_node' => true
+                ))
             ->end()
         ;
+
         parent::configureFormFields($formMapper);
+
+        if (null === $this->getParentFieldDescription()) {
+
+            // Add the choice for the node links "target"
+            $formMapper
+                ->with('form.group_general')
+                    ->add('linkType', 'choice_field_mask', array(
+                        'choices' => array_combine(
+                            $this->contentAwareFactory->getLinkTypes(),
+                            $this->contentAwareFactory->getLinkTypes()
+                        ),
+                        'map' => array(
+                            'route' => array('route'),
+                            'uri' => array('uri'),
+                            'content' => array('content', 'doctrine_phpcr_odm_tree'),
+                        ),
+                        'empty_value' => 'auto',
+                        'required' => false
+                    ))
+                    //->add('route', 'text', array('required' => false))
```

Comment:
ok, cool, thanks for the detailed explanation!
but then we should remove this commented out line before merging.